ocaml/xapi/xapi_pusb.ml

```diff
@@ -122,16 +122,17 @@ let set_passthrough_enabled ~__context ~self ~value =
        Db.PUSB.set_passthrough_enabled ~__context ~self ~value
      | false ->
        try
-         let attached = Db.PUSB.get_attached ~__context ~self in
-         (* If the USB is passthroughed to vm, need to unplug it firstly*)
-         if attached <> Ref.null then begin
-           let vm = Db.VUSB.get_VM ~__context ~self:attached in
-           raise (Api_errors.Server_error(Api_errors.usb_already_attached, [Ref.string_of self; Ref.string_of vm]))
-         end;
          let usb_group = Db.PUSB.get_USB_group ~__context ~self in
          let vusbs = Db.USB_group.get_VUSBs ~__context ~self:usb_group in
-         (* If vusb has been created, need to destroy it. *)
-         List.iter (fun vusb -> Xapi_vusb.destroy ~__context ~self:vusb) vusbs;
+         (* If the USB is passthroughed to vm, need to unplug it firstly*)
+         if vusbs <> [] then begin
+           let currently_attached = Db.VUSB.get_currently_attached ~__context ~self:(List.hd vusbs) in
+           if currently_attached then
+             let vm = Db.VUSB.get_VM ~__context ~self:(List.hd vusbs) in
```

Comment:
And this one. Though we could avoid both by changing the
```ocaml
if vusbs <> [] then begin
```
to
```ocaml
match vusbs with
| [] -> ()
| [vusb] -> begin
```

Reply:
You can also protect the "exactly one" case adding `| _ :: _ :: _ -> error "too many vusb on the pusb"` with your second suggestion

Reply:
updated.